### PR TITLE
allow running JS tests for a single plugin in tests:run-js

### PR DESCRIPTION
**Pull request by diosmosis**
### Description:

Adds a `--plugin=$pluginName` option to tests:run-js. This is to fix the JS tests in a premium plugin.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by diosmosis** [comment]
@matomo-org/core-reviewers can someone take a look at this PR?

File: tests/javascript/testrunnerPhantom.js [hunk 1]
```diff
@@ -23,7 +23,23 @@
...
+var baseUrl = system.args[1] || 'http://localhost/tests/javascript/';
+
+function getPluginArg() {
+  for (var i = 0; i < system.args.length; ++i) {
+    if (/--plugin=(.*?)/.test(system.args[i])) {
+      return system.args[i].split('=', 2)[1];
+    }
+  }
+  return null;
+}
```
Comment (michalkleiner) [inline tests/javascript/testrunnerPhantom.js:35]: Is it possible to use the same approach as in the nodeRunner or was there any specific reason to use a different iteration mechanism here?

```suggestion
function getPluginArg() {
  const pluginArg = system.args.find((arg) => /--plugin=(.*?)/.test(arg));
  return pluginArg ? pluginArg.split('=', 2)[1] : null;
}
```

**Comment by michalkleiner** [comment]
The changes look ok, left one suggestion. Not sure I see how adding the option to run tests for a single plugin instead of all of them fixes an issue for a plugin. Do the tests fail there if run as part of the overall set and not when run in isolation? That could be an indication of a hidden problem somewhere else.

**Comment by bx80** [comment]
It's useful to be able to run the tests for just one plugin when debugging an issue locally, I think the idea here is to help with fixing the tests in a premium plugin by not having to run all the tests each time :slightly_smiling_face:

**Comment by michalkleiner** [comment]
Yep, definitely in support of the change, just wasn't 100% sure on the PR description.

File: tests/javascript/testrunnerPhantom.js (hunk 1 above)
Comment (diosmosis) [reply tests/javascript/testrunnerPhantom.js:35]: phantomjs does not support modern ecmascript syntax and I'm not sure if it has ES5 array functions.

Comment (michalkleiner) [reply tests/javascript/testrunnerPhantom.js:35]: Thanks, that's definitely a good reason to keep it the old way.

**Review by michalkleiner — APPROVED** [review]
Looks ok to me but I won't merge it yet so @sgiehl and @bx80 can have a look as well.
